mkdocs/config/config_options.py

```diff
@@ -199,6 +200,58 @@ def run_validation(self, value: object) -> list[T]:
         return [fake_config[k] for k in fake_keys]
 
 
+class DictOfItems(Generic[T], BaseConfigOption[Dict[str, T]]):
+    """
+    Validates a dict of items. Keys are always strings.
+
+    E.g. for `config_options.DictOfItems(config_options.Type(int))` a valid item is `{"a": 1, "b": 2}`.
+    """
+
+    required: Union[bool, None] = None  # Only for subclasses to set.
```

Comment:
Maybe the branch was out of date when you copied this part :)
```suggestion
    required: bool | None = None  # Only for subclasses to set.
```

Reply:
Yeah most probably, thanks for fixing it!